### PR TITLE
feat: index carbon components

### LINKED ISSUE #515: Index Carbon Components

### PR DESCRIPTION
**Pull request by andreancardona**
Closes https://github.com/carbon-design-system/carbon-platform/issues/515

### CODE REVIEW

File: packages/carbon-components/carbon.yml [hunk 1]
```diff
@@ -0,0 +1,416 @@
+# yaml-language-server: $schema=https://unpkg.com/@carbon-platform/schemas@v1/carbon-resources.schema.json
+library:
+  id: carbon-components
+  name: Carbon Components
+  description: Vanilla JavaScript Carbon components
+  externalDocsUrl: https://the-carbon-components.netlify.app/
```
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
  externalDocsUrl: https://the-carbon-components.netlify.app
```

File: packages/carbon-components/carbon.yml [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+assets:
+    accordion:
+      status: stable
+      externalDocsUrl: https://www.carbondesignsystem.com/components/accordion/usage
+      framework: vanilla
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://react.carbondesignsystem.com/?path=/docs/components-accordion
```
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]: i
```suggestion
          url: https://the-carbon-components.netlify.app/?nav=accordion
```

File: packages/carbon-components/carbon.yml [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+          action: link
+          url: https://the-carbon-components.netlify.app/?nav=content-switcher
+    copy-button:
+      status: stable
+      framework: vanilla
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://react.carbondesignsystem.com/?path=/docs/components-copybutton
```
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
```

File: packages/carbon-components/carbon.yml [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+      status: stable
+      type: component
+      platform: web
+      # TODO: add thumbnail
+      thumbnailPath: ''
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://the-carbon-components.netlify.app/?nav=loading
```
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
```

File: packages/carbon-components/carbon.yml [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+      status: stable
+      framework: vanilla
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://the-carbon-components.netlify.app/?nav=pagination-nav
+    popover:
+      status: stable
+      framework: react
```
Comment (francinelucca) [inline packages/carbon-components/carbon.yml]:
```suggestion
      framework: vanilla
```

**Review by francinelucca — CHANGES_REQUESTED** [review]

**Review by francinelucca — COMMENTED** [review]
LGTM, can see 46 components as expected: 
image

And library info:
image

This should be good to go once we resolve remaining TODOs (get final thumbnails and tag approval) @andreancardona  👍 

Let's also confirm treeview status @andreancardona since it's unstable on styles and it's not present in storybook

File: packages/carbon-components/carbon.yml [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+      type: component
+      platform: web
+      # TODO: add thumbnail
+      thumbnailPath: '' 
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://the-carbon-components.netlify.app/?nav=list-box
+      # TODO: @aubrey check tag
```
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Approved 👍

File: packages/carbon-components/carbon.yml [hunk 7]
```diff
@@ -0,0 +1,406 @@
...
+          url: https://the-carbon-components.netlify.app/?nav=loading
+    menu:
+      name: Menu
+      framework: vanilla
+      status: stable
+      type: component
+      platform: web
+      # TODO: add thumbnail
+      thumbnailPath: ''
+      # TODO: @aubrey check tag
```
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Remove this component.

File: packages/carbon-components/carbon.yml [hunk 8]
```diff
@@ -0,0 +1,406 @@
...
+      type: component
+      platform: web
+      # TODO: add thumbnail
+      thumbnailPath: '' 
+      demoLinks:
+        - type: storybook
+          name: Demo
+          action: link
+          url: https://the-carbon-components.netlify.app/?nav=skeleton
+      # TODO: @aubrey check tag
```
Comment (aubrey-oneal) [inline packages/carbon-components/carbon.yml]: Approved 👍

**Review by aubrey-oneal — CHANGES_REQUESTED** [review]
Just remove the menu component.

**Review by francinelucca — APPROVED** [review]
Lib and comps being pulled correctly:
image
LGTM

**Pull request closed without merging**